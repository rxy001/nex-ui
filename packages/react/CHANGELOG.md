# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
