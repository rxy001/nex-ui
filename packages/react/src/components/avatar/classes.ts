import { generateUtilityClasses } from '../utils'

export const avatarClasses = generateUtilityClasses('nui-avatar', [
  'root',

  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-xl',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'outlined',

  'img',
])

export const avatarGroupClasses = generateUtilityClasses('nui-avatar-group', [
  'root',
  'surplus',
])
