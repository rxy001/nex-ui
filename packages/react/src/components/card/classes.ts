import { generateUtilityClasses } from '../utils'

export const cardClasses = generateUtilityClasses('nui-card', [
  'root',
  'blurred',
  'hoverable',

  'shadow-xs',
  'shadow-sm',
  'shadow-md',
  'shadow-lg',
  'shadow-xl',

  'radius-none',
  'radius-sm',
  'radius-md',
  'radius-lg',
])

export const cardHeaderClasses = generateUtilityClasses('nui-card-header', [
  'root',
  'content',
  'title',
  'subtitle',
])

export const cardBodyClasses = generateUtilityClasses('nui-card-body', ['root'])

export const cardFooterClasses = generateUtilityClasses('nui-card-footer', [
  'root',
])

export const cardActionAreaClasses = generateUtilityClasses(
  'nui-card-action-area',
  ['root'],
)
