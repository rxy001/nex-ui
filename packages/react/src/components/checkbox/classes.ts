import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const checkboxClasses = generateUtilityClasses('nui-checkbox', [
  // ---------root---------
  'root',

  // "sm" | "md" | "lg" | "full" | "none"
  'radius-sm',
  'radius-md',
  'radius-lg',
  'radius-full',
  'radius-none',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  'indeterminate',
  // ---------root---------

  'input',
  'label',
  'icon',
])

export const checkboxGroupClasses = generateUtilityClasses(
  'nui-checkbox-group',
  [
    'root',
    'label',
    'wrapper',
    'orientation-vertical',
    'orientation-horizontal',
  ],
)
