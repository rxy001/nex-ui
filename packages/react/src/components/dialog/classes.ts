import { generateUtilityClasses } from '../utils'

export const dialogClasses = generateUtilityClasses('nui-dialog', [
  'root',
  'open',
  'backdrop',
])

export const dialogContentClasses = generateUtilityClasses(
  'nui-dialog-content',
  [
    'root',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'full-screen',

    'placement-top',
    'placement-bottom',
    'placement-center',

    'scroll-inside',
    'scroll-outside',

    'paper',
    'close-button',
  ],
)

export const dialogHeaderClasses = generateUtilityClasses('nui-dialog-header', [
  'root',
])

export const dialogFooterClasses = generateUtilityClasses('nui-dialog-footer', [
  'root',
])

export const dialogBodyClasses = generateUtilityClasses('nui-dialog-body', [
  'root',
])
