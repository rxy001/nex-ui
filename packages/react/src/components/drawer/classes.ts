import { generateUtilityClasses } from '../utils'

export const drawerClasses = generateUtilityClasses('nui-drawer', [
  'root',
  'open',
  'backdrop',
])

export const drawerContentClasses = generateUtilityClasses(
  'nui-drawer-content',
  [
    'root',

    'placement-top',
    'placement-right',
    'placement-bottom',
    'placement-left',

    'size-xs',
    'size-sm',
    'size-md',
    'size-lg',
    'size-xl',
    'size-full',

    'paper',
    'close-button',
  ],
)

export const drawerHeaderClasses = generateUtilityClasses('nui-drawer-header', [
  'root',
])

export const drawerFooterClasses = generateUtilityClasses('nui-drawer-footer', [
  'root',
])

export const drawerBodyClasses = generateUtilityClasses('nui-drawer-body', [
  'root',
])
