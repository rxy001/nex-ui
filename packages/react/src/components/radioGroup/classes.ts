import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const radioClasses = generateUtilityClasses('nui-radio', [
  // ---------root---------
  'root',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'checked',
  'disabled',
  // ---------root---------

  'input',
  'label',
  'dot',
])

export const radioGroupClasses = generateUtilityClasses('nui-radio-group', [
  'root',
  'label',
  'wrapper',
  'orientation-vertical',
  'orientation-horizontal',
])
