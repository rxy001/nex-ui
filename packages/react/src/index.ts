import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
