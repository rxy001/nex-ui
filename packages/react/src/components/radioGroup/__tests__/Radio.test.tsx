import { act, fireEvent } from '@testing-library/react'
import {
  testComponentStability,
  testRootClassName,
  testColorClasses,
  testSizeClasses,
  renderWithNexUIProvider,
} from '~/tests/shared'
import { createRef } from 'react'
import { Radio, RadioGroup } from '../index'
import { radioClasses } from '../classes'

describe('Radio', () => {
  testComponentStability(<Radio value='1'>Option 1</Radio>)

  testRootClassName(<Radio className='test-class' />, 'test-class')

  testColorClasses(<Radio>Radio</Radio>, radioClasses)

  testSizeClasses(<Radio>Radio</Radio>, radioClasses)

  it('should render with default props', () => {
    const { container, getByRole } = renderWithNexUIProvider(
      <Radio>Radio</Radio>,
    )

    const radioRoot = container.firstElementChild
    expect(radioRoot).toHaveClass(radioClasses.root)
    expect(radioRoot).toHaveClass(radioClasses['size-md'])
    expect(radioRoot).toHaveClass(radioClasses['color-blue'])

    expect(radioRoot).not.toHaveClass(radioClasses['color-green'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-cyan'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-orange'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-pink'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-purple'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-yellow'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-red'])
    expect(radioRoot).not.toHaveClass(radioClasses['color-gray'])
    expect(radioRoot).not.toHaveClass(radioClasses['size-sm'])
    expect(radioRoot).not.toHaveClass(radioClasses['size-lg'])
    expect(radioRoot).not.toHaveClass(radioClasses.checked)
    expect(radioRoot).not.toHaveClass(radioClasses.disabled)
    expect(radioRoot).toMatchSnapshot()

    const radio = getByRole('radio')
    expect(radio).toHaveClass(radioClasses.input)
    expect(radio.nextElementSibling).toHaveClass(radioClasses.dot)
  })

  it("should forward ref to Radio's input element", () => {
    const ref = createRef<HTMLInputElement>()
    const { getByRole } = renderWithNexUIProvider(<Radio ref={ref} />)
    expect(ref.current).toBe(getByRole('radio'))
  })

  it("should render Radio's label with text children", () => {
    const { getByText } = renderWithNexUIProvider(<Radio>Test Label</Radio>)
    expect(getByText('Test Label')).toBeInTheDocument()
  })

  it('should render unchecked radio by default', () => {
    const { getByRole } = renderWithNexUIProvider(<Radio />)

    expect(getByRole('radio')).not.toBeChecked()
  })

  it('should render checked radio by defaultChecked prop', () => {
    const { getByRole } = renderWithNexUIProvider(<Radio defaultChecked />)

    expect(getByRole('radio')).toBeChecked()
  })

  it('should render checked radio when checked prop is true', () => {
    const { getByRole, container } = renderWithNexUIProvider(<Radio checked />)

    expect(getByRole('radio')).toBeChecked()
    expect(container.firstElementChild).toHaveClass(radioClasses.checked)
  })

  it('should switch checked state and call onCheckedChange when clicked', async () => {
    const onCheckedChange = jest.fn()
    const { getByRole } = renderWithNexUIProvider(
      <Radio onCheckedChange={onCheckedChange} />,
    )

    const radio = getByRole('radio')
    expect(radio).not.toBeChecked()

    await act(async () => {
      fireEvent.click(radio)
    })

    expect(radio).toBeChecked()
    expect(onCheckedChange).toHaveBeenCalledWith(true)
  })

  it('should forward classes to root, input, dot and label slots', () => {
    const classes = {
      root: 'test-class-root',
      input: 'test-class-input',
      dot: 'test-class-dot',
      label: 'test-class-label',
    }

    const { container } = renderWithNexUIProvider(
      <Radio classes={classes}>Radio</Radio>,
    )

    const radioRoot = container.querySelector(`.${radioClasses.root}`)
    const radioLabel = container.querySelector(`.${radioClasses.label}`)
    const radioDot = container.querySelector(`.${radioClasses.dot}`)
    const radio = container.querySelector(`.${radioClasses.input}`)

    expect(radioRoot).toHaveClass(classes.root)
    expect(radioDot).toHaveClass(classes.dot)
    expect(radioLabel).toHaveClass(classes.label)
    expect(radio).toHaveClass(classes.input)
  })

  it('should forward slotProps to root, dot and label slots', () => {
    const { container } = renderWithNexUIProvider(
      <Radio
        slotProps={{
          root: { className: 'test-class-root' },
          label: { className: 'test-class-label' },
          dot: { className: 'test-class-dot' },
        }}
      >
        Radio
      </Radio>,
    )

    const radioRoot = container.querySelector(`.${radioClasses.root}`)
    const radioLabel = container.querySelector(`.${radioClasses.label}`)
    const radioIcon = container.querySelector(`.${radioClasses.dot}`)

    expect(radioRoot).toHaveClass('test-class-root')
    expect(radioLabel).toHaveClass('test-class-label')
    expect(radioIcon).toHaveClass('test-class-dot')
  })

  it('should disable the radio when disabled prop is true', () => {
    const { getByRole, container } = renderWithNexUIProvider(<Radio disabled />)

    const radio = getByRole('radio')
    const radioRoot = container.firstElementChild

    expect(radio).toBeDisabled()
    expect(radioRoot).toHaveClass(radioClasses.disabled)
    expect(radioRoot).toHaveStyleRule('pointer-events', 'none')
  })

  it("should warn when Radio's checked is set within RadioGroup", () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()

    renderWithNexUIProvider(
      <RadioGroup>
        <Radio value='orange' checked>
          orange
        </Radio>
      </RadioGroup>,
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Radio: The RadioGroup is being used, `checked` will be ignored. Use the `value` of the RadioGroup instead.',
    )

    consoleSpy.mockRestore()
  })

  it("should warn when Radio's defaultChecked is set within RadioGroup", () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation()

    renderWithNexUIProvider(
      <RadioGroup>
        <Radio value='orange' defaultChecked>
          orange
        </Radio>
      </RadioGroup>,
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Radio: The RadioGroup is being used, `defaultChecked` will be ignored. Use the `defaultValue` of the RadioGroup instead.',
    )

    consoleSpy.mockRestore()
  })

  it("should error when Radio's value is not set within RadioGroup", () => {
    const consoleSpy = jest.spyOn(console, 'error').mockImplementation()

    renderWithNexUIProvider(
      <RadioGroup>
        <Radio>orange</Radio>
      </RadioGroup>,
    )

    expect(consoleSpy).toHaveBeenCalledWith(
      '[Nex UI] Radio: The `value` prop is required when using Radio inside a RadioGroup',
    )

    consoleSpy.mockRestore()
  })

  it('should have data-focus-visible attribute when focused', async () => {
    const { getByRole, user } = renderWithNexUIProvider(
      <Radio as='div'>Radio</Radio>,
    )

    const radio = getByRole('radio')

    expect(radio).not.toHaveAttribute('data-focus-visible')

    await user.tab()
    expect(document.activeElement).toBe(radio)
    expect(radio).toHaveAttribute('data-focus-visible', 'true')
  })

  describe('Accessibility', () => {
    it('should have type="radio" by default', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('type', 'radio')
    })

    it('should not apply role="radio" when rendered as an input', () => {
      const { container } = renderWithNexUIProvider(
        <Radio as='input'>Radio</Radio>,
      )
      const radio = container.querySelector(`.${radioClasses.input}`)
      expect(radio).not.toHaveAttribute('role', 'radio')
    })

    it('should apply role="radio" when rendered as a non-input element', () => {
      const { container } = renderWithNexUIProvider(
        <Radio as='div'>Radio</Radio>,
      )

      const radio = container.querySelector(`.${radioClasses.input}`)
      expect(radio).toHaveAttribute('role', 'radio')
    })

    it('should apply checked attribute when checked', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio checked />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('checked', '')
      expect(radio).not.toHaveAttribute('aria-checked', 'true')
    })

    it('should apply aria-checked="true" to non-input elements when checked', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio as='div' checked />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-checked', 'true')
      expect(radio).not.toHaveAttribute('checked')
    })

    it('should apply aria-checked="false" to non-input elements when unchecked', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio as='div' />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-checked', 'false')
      expect(radio).not.toHaveAttribute('checked')
    })

    it('should apply disabled attribute when disabled', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio disabled />)
      const radio = getByRole('radio')
      expect(radio).toBeDisabled()
      expect(radio).not.toHaveAttribute('aria-disabled')
    })

    it('should apply aria-disabled="true" to non-input elements when disabled', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio as='div' disabled />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-disabled', 'true')
      expect(radio).not.toBeDisabled()
    })

    it('should have tabIndex=0 by default', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('tabIndex', '0')
    })

    it('should apply tabIndex=-1 when disabled', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio disabled />)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('tabIndex', '-1')
    })

    it('should apply aria-label when children is a string', () => {
      const { getByRole } = renderWithNexUIProvider(<Radio>Radio Label</Radio>)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-label', 'Radio Label')
    })

    it('should apply aria-label when aria-label prop is provided', () => {
      const { getByRole } = renderWithNexUIProvider(
        <Radio aria-label='Custom Label'>Radio</Radio>,
      )
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-label', 'Custom Label')
    })

    it('should apply aria-labelledby when aria-labelledby prop is provided', () => {
      const { getByRole } = renderWithNexUIProvider(
        <Radio aria-labelledby='label-id'>
          <span id='label-id'>Radio Label</span>
        </Radio>,
      )
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-labelledby', 'label-id')
      expect(radio).not.toHaveAttribute('aria-label')
    })

    it('should apply aria-labelledby="label-id" by default when children is a string', () => {
      const { getByRole, container } = renderWithNexUIProvider(
        <Radio>Radio Label</Radio>,
      )
      const label = container.querySelector(`.${radioClasses.label}`)
      const radio = getByRole('radio')
      expect(radio).toHaveAttribute('aria-labelledby', label!.id)
    })

    it('should check non-input elements when clicked', async () => {
      const { container, getByRole } = renderWithNexUIProvider(
        <Radio as='div'>Radio</Radio>,
      )
      const radioRoot = container.firstElementChild!
      const radio = getByRole('radio')

      expect(radioRoot).not.toHaveClass(radioClasses.checked)
      expect(radio).toHaveAttribute('aria-checked', 'false')

      await act(async () => {
        fireEvent.click(radio)
      })

      expect(radioRoot).toHaveClass(radioClasses.checked)
      expect(radio).toHaveAttribute('aria-checked', 'true')
    })

    it('should check non-input elements when space is pressed', async () => {
      const onChange = jest.fn()

      const { user, getByRole } = renderWithNexUIProvider(
        <Radio as='span' onCheckedChange={onChange}>
          Focusable Span
        </Radio>,
      )

      const span = getByRole('radio')
      await user.tab()
      expect(document.activeElement).toBe(span)

      await user.keyboard(' ')
      expect(span).toHaveAttribute('aria-checked', 'true')
      expect(onChange).toHaveBeenCalledWith(true)
    })
  })
})
