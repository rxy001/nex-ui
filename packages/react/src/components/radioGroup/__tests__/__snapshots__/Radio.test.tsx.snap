// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render with default props 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  padding: var(--nui-spaces-1-5);
  height: var(--nui-sizes-8);
}

.emotion-1 {
  position: absolute;
  inset: 0;
  opacity: 0;
  margin: 0;
  cursor: inherit;
  z-index: 1;
}

.emotion-1:is(:hover, [data-hover=true]):not(:disabled, [data-disabled=true])+* {
  border-color: var(--nui-colors-blue-secondary);
}

.emotion-1:is(:focus-visible, [data-focus-visible=true]):not(:disabled, [data-disabled=true])+* {
  outline: var(--nui-borders-focus-visible-outline);
  outline-offset: var(--nui-spaces-0-5);
}

.emotion-2 {
  border-radius: var(--nui-radii-full);
  border: var(--nui-borders-md);
  border-color: var(--nui-colors-gray-highlight);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  box-sizing: border-box;
  -webkit-transition: var(--nui-transitions-colors);
  transition: var(--nui-transitions-colors);
  width: var(--nui-sizes-5);
  height: var(--nui-sizes-5);
}

.emotion-2::before {
  content: "";
  display: block;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  border-radius: var(--nui-radii-full);
  background-color: var(--nui-colors-blue-primary);
  -webkit-transition: var(--nui-transitions-all);
  transition: var(--nui-transitions-all);
  width: var(--nui-sizes-2);
  height: var(--nui-sizes-2);
}

.emotion-3 {
  -webkit-margin-start: var(--nui-spaces-2);
  margin-inline-start: var(--nui-spaces-2);
  font-size: var(--nui-font-sizes-lg);
}

<label
  class="nui-radio-root nui-radio-size-md nui-radio-color-blue emotion-0"
>
  <input
    aria-label="Radio"
    aria-labelledby="«re»"
    class="nui-radio-input emotion-1"
    tabindex="0"
    type="radio"
  />
  <span
    class="nui-radio-dot emotion-2"
  />
  <span
    class="nui-radio-label emotion-3"
    id="«re»"
  >
    Radio
  </span>
</label>
`;
